### ISSUE TITLE
cargo-binstall for Dioxus CLI fails if release tag is not consitent

### ISSUE THREAD

**Issue by esimkowitz** [DioxusLabs/dioxus#1430]
**Problem**

The current release of the Dioxus CLI uses a different release tag naming scheme than that used for the previous GitHub Releases. This breaks the pattern set in the [Cargo.toml](https://github.com/DioxusLabs/dioxus/blob/master/packages/cli/Cargo.toml), meaning cargo-binstall cannot find the binaries and falls back to the standard cargo install procedure.

**Steps To Reproduce**

- Create a new GitHub Actions workflow with the following:
```yml
name: Test cargo-binstall

on:
  push:
    branches: ["main"]
  workflow_dispatch:

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout repo
        uses: actions/checkout@v3
      
      # Install toolchain
      - name: Install Rust Toolchain
        uses: dtolnay/rust-toolchain@master
        with:
          toolchain: stable
          targets: wasm32-unknown-unknown
      - name: Install Dioxus CLI
        uses: taiki-e/install-action@v1
        with:
          tool: dioxus-cli@0.4.1
```
- Check this into the main branch of a dummy repository and the action should automatically run
- You will see that rather than install the pre-built binaries, the action compiles from source code.

**Expected behavior**

When running cargo-binstall or a GitHub Action like [taiki-e/install-action](https://github.com/taiki-e/install-action) which uses cargo-binstall under the hood to install Dioxus CLI v0.4.1 or later, the pre-built binaries should be installed rather than re-compiling the CLI from source. 

**Screenshots**

**Environment:**
 - Dioxus version: v0.4.1
 - Rust version: 1.71.0
 - OS info: Ubuntu 22.04
 - App platform: web

**Questionnaire**

- [ ] I'm interested in fixing this myself but don't know where to start
- [x] I would like to fix and I have a solution
- [ ] I don't have time to fix this right now, but maybe later

**Comment by esimkowitz** [comment]
To get around this, the [Build CLI for release workflow](https://github.com/DioxusLabs/dioxus/actions/workflows/cli_release.yml) could update the value of the `pkg-url` with the exact title of the release, rather than using a pattern. It could then commit that change back to the release tag.

**Comment by esimkowitz** [comment]
I have a potential fix ready: #1431
